src/main/java/org/mockito/internal/creation/SuspendMethod.java

```diff
@@ -0,0 +1,18 @@
+package org.mockito.internal.creation;
+
+import java.util.Arrays;
+
+/**
+ * Utilities for Kotlin Continuation-Passing-Style suspending function, detecting and trimming last hidden parameter.
+ * See <a href="https://github.com/Kotlin/kotlin-coroutines/blob/master/kotlin-coroutines-informal.md#continuation-passing-style">Design docs for details</a>.
+ */
+public class SuspendMethod {
```

Comment:
Thank you very much for adding comprehensive Javadoc with the context and link to Kotlin docs!